deployments/charts/kuma/templates/pre-upgrade-install-missing-crds-job.yaml

```diff
@@ -113,15 +114,18 @@ spec:
       {{ toYaml . | nindent 8 }}
       {{- end }}
       restartPolicy: OnFailure
-      {{- if .Values.hooks.podSecurityContext }}
       securityContext:
+        runAsNonRoot: true
+      {{- if .Values.hooks.podSecurityContext }}
       {{ toYaml .Values.hooks.podSecurityContext | trim | nindent 8 }}
       {{- end }}
       containers:
         - name: pre-upgrade-job
           image: {{ include "kubectl.formatImage" (dict "image" .Values.kubectl.image "root" $) | quote }}
-          {{- if .Values.hooks.containerSecurityContext }}
           securityContext:
+            runAsUser: 65534 # the nobody user in the default image
+            readOnlyRootFilesystem: true
+          {{- if .Values.hooks.containerSecurityContext }}
```

Comment:
Can you add a test that introduces a custom security context which overrides runAsUser for example? I feel like the override will create invalid yaml and you may need to use [merge](https://helm.sh/docs/chart_template_guide/function_list/#merge-mustmerge) instead

Reply:
done